src/Microsoft.TestPlatform.Common/Telemetry/MetricsCollection.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.Telemetry
+{
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    /// <inheritdoc />
+    /// <summary>
+    /// This Class will collect Metrics.
+    /// </summary>
+    public class MetricsCollection : IMetricsCollection
+    {
+        private Dictionary<string, string> metricDictionary;
+
+        /// <summary>
+        /// The Metrics Collector
+        /// </summary>
+        public MetricsCollection()
+        {
+            this.metricDictionary = new Dictionary<string, string>();
+        }
+
+        /// <summary>
+        /// Add The Metrics
+        /// </summary>
+        /// <param name="metric"></param>
+        /// <param name="value"></param>
+        public void Add(string metric, string value)
+        {
+            ValidateArg.NotNull(metric, "metric");
```

Comment:
This could through exception and therefore, could result in interference with Test Execution.
Test execution should not fail in case of any telemetry related exception.